code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm

```diff
@@ -615,24 +637,32 @@
 	taste_description = "bitterness"
 	metabolism = REM * 2
 	unaffected_species = IS_DIONA | IS_MACHINE
-	var/datum/modifier/modifier
 
-/datum/reagent/toxin/berserk/affect_blood(var/mob/living/carbon/M, var/removed)
+/decl/reagent/toxin/berserk/initial_effect(var/mob/living/carbon/human/H, var/alien, var/holder)
+	. = ..()
+	if(!istype(H))
+		return .
+	H.berserk_start()
+
+/decl/reagent/toxin/berserk/final_effect(var/mob/living/carbon/human/H, var/alien, var/holder)
+	. = ..()
+	if(!istype(H))
+		return .
+	H.berserk_stop()
```

Comment:
These are fine, but I feel like it is more readable if you negate the if condition:

```suggestion
/decl/reagent/toxin/berserk/initial_effect(var/mob/living/carbon/human/H, var/alien, var/holder)
	. = ..()
	if(istype(H))
		H.berserk_start()

/decl/reagent/toxin/berserk/final_effect(var/mob/living/carbon/human/H, var/alien, var/holder)
	. = ..()
	if(istype(H))
		H.berserk_stop()
```